### PR TITLE
ci: update to ansys actions v8 and docs theme (static search)

### PR DESCRIPTION
**Pull request by RobPasMue**
## Description
Updating to actions (v8), Ansys sphinx theme to 1.1 and removing Meilisearch in favor of static search. Also putting an upper limit on protobug 5.28 due to annoying warnings.

## Issue linked
None

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -75,7 +75,7 @@ tests-minimal = [
     "pytest-xvfb==3.0.0",
 ]
 doc = [
-    "ansys-sphinx-theme[autoapi]==1.0.11",
+    "ansys-sphinx-theme[autoapi]==1.1.0",
```
Comment (RobPasMue) [inline pyproject.toml:78]:
```suggestion
    "ansys-sphinx-theme[autoapi]==1.1.1",
```

File: doc/source/conf.py [hunk 2]
```diff
@@ -160,15 +160,13 @@ def intersphinx_pyansys_geometry(switcher_version: str):
...
-        "index_uids": {
-            f"pyansys-geometry-v{get_version_match(__version__).replace('.', '-')}": "PyAnsys-Geometry",  # noqa: E501
-        },
-    },
     "ansys_sphinx_theme_autoapi": {
         "project": project,
     },
+    "static_search": {
+        "threshold": 0.5,
+        "min_chars_for_search": 2,
```
Comment (Revathyvenugopal162) [inline doc/source/conf.py:168]:
```suggestion
        "min_chars_for_search": 2,
        "ignoreLocation": True,
```
For smaller libraries with fewer documents, we can use this option without impacting search time while achieving better results. Feel free to disregard, up to you: [Fuse.js - `ignoreLocation` option](https://www.fusejs.io/api/options.html#ignorelocation).

Comment (RobPasMue) [reply doc/source/conf.py:168]: Cool, didn't know that! Thanks @Revathyvenugopal162!

Comment (Revathyvenugopal162) [reply doc/source/conf.py:168]: we can use any options from Fuse.js in our theme options depending on the size of docs.

**Review by SMoraisAnsys — APPROVED** [review]
LGTM
Just wondering, what's the reason for the protobuf changes ?

**Comment by RobPasMue** [comment]
> LGTM Just wondering, what's the reason for the protobuf changes ?

Let me explain them later in our weekly meeting. Mostly this annoying warning that you get if you use 5.28.2 and the latest grpcio-health-check versions https://github.com/ansys/pyansys-geometry/actions/runs/11119940649/job/30896029319#step:7:1777 

There are issues and PRs for solving this on the grpc repo but it's very annoying and users have already complained about getting these warnings. Bottom line - it's not a PyAnsys Geometry issue, it's a dependency issue... but we are impacted.

**Pull request merged**